### PR TITLE
Support SENTRYCLI_CDNURL env

### LINKED ISSUE #150: Support SENTRYCLI_CDNURL env
It's hard to complete the sentry cli download phase because of internet censorship in China. With the support of SENTRYCLI_CDNURL env, users who can not download directly from Sentry CDN can download from alternative place.

Sentry npm package support this env, link: https://docs.sentry.io/product/cli/installation/#available-installation-options.

I can make a PR about this.

### PR DESCRIPTION
**Pull request by fishead**
## :scroll: Description

Add the option to provide an alternative place to download sentry-cli binary

## :bulb: Motivation and Context

close #150

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by buenaflor** [comment]
Sorry for the delay, I will be reviewing this soon

File: README.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ ignore_missing=true
 | ignore_missing | Ignore missing commits previously used in the release | default: false | no | - |
 | bin_dir | The folder where the plugin downloads the sentry-cli binary | .dart_tool/pub/bin/sentry_dart_plugin (string) | no | - |
 | bin_path | Path to the sentry-cli binary to use instead of downloading. Make sure to use the correct version. | null (string) | no | - |
+| sentry_cli_cdn_url | Alternative place to download sentry-cli | default: https://downloads.sentry-cdn.com/sentry-cli | no | SENTRYCLI_CDNURL |
```
Comment (denrase) [inline README.md:116]: I'd call the env var `SENTRY_CLI_CDN_URL` as it is easier to read and also mirrors the format of `sentry_cli_cdn_url`

File: lib/src/configuration.dart [hunk 2]
```diff
@@ -90,6 +90,9 @@ class Configuration {
   /// downloaded. Please make sure to use the matching version.
   late String? binPath;
 
+  /// Place to download sentry-cli
```
Comment (denrase) [inline lib/src/configuration.dart:93]: We could add the default value in the comment, same as for the other properties.

File: lib/src/configuration.dart [hunk 3]
```diff
@@ -146,6 +149,8 @@ class Configuration {
     logLevel = configValues.logLevel; // or env. var. SENTRY_LOG_LEVEL
     binDir = configValues.binDir ?? '.dart_tool/pub/bin/sentry_dart_plugin';
     binPath = configValues.binPath;
+    sentryCliCdnUrl = configValues.sentryCliCdnUrl ??
+        'https://downloads.sentry-cdn.com/sentry-cli';
```
Comment (denrase) [inline lib/src/configuration.dart:153]: When we provide the default path here, we'd never need the value from `_sources.dart`, as we always call the constructor where we pass this prefix.

We can probably just remove it from "_sources.dart" and just make `_formatDownloadUrl(String prefix)` public and remove `_formatDownloadUrl` getter.

File: test/configuration_test.dart [hunk 4]
```diff
@@ -199,6 +208,10 @@ void main() {
       );
       expect(sut.waitForProcessing, false);
       expect(sut.binDir, '.dart_tool/pub/bin/sentry_dart_plugin');
+      expect(
+        sut.sentryCliCdnUrl,
+        'https://downloads.sentry-cdn.com/sentry-cli',
```
Comment (denrase) [inline test/configuration_test.dart:213]: ❤️

**Review by denrase — CHANGES_REQUESTED** [review]
Thank you for contributing @fishead! 🙇

File: README.md (hunk 1 above)
Comment (fishead) [reply README.md:116]: The env var name `SENTRYCLI_CDNURL` comes from the official doc https://docs.sentry.io/cli/installation/#downloading-from-a-custom-source, I've used it in JS project, so I think different sentry SDK can share the same env name.

Should we support both `SENTRY_CLI_CDN_URL` and `SENTRYCLI_CDNURL`, and make `SENTRYCLI_CDNURL` as fallback, or just `SENTRY_CLI_CDN_URL`?

File: lib/src/configuration.dart (hunk 3 above)
Comment (fishead) [reply lib/src/configuration.dart:153]: I have the same idea when I coding the PR.

File: lib/src/configuration.dart (hunk 2 above)
Comment (fishead) [reply lib/src/configuration.dart]: done.

File: lib/src/configuration.dart (hunk 3 above)
Comment (fishead) [reply lib/src/configuration.dart:155]: done.

**Review by denrase — COMMENTED** [review]
Nice, thank you for contributing @fishead 🙇

**Review by denrase — APPROVED** [review]
Nice, thank you for contributing @fishead 🙇

**Pull request merged**